frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -630,24 +643,28 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 				return nil, parser.WithLocation(err, d.stage.Location)
 			}
 		}
+
 		d.state = d.state.Network(opt.NetworkMode)
+
 		opt := dispatchOpt{
-			allDispatchStates: allDispatchStates,
-			metaArgs:          optMetaArgs,
-			buildArgValues:    opt.BuildArgs,
-			shlex:             shlex,
-			buildContext:      llb.NewState(buildContext),
-			proxyEnv:          proxyEnv,
-			cacheIDNamespace:  opt.CacheIDNamespace,
-			buildPlatforms:    platformOpt.buildPlatforms,
-			targetPlatform:    platformOpt.targetPlatform,
-			extraHosts:        opt.ExtraHosts,
-			shmSize:           opt.ShmSize,
-			ulimit:            opt.Ulimits,
-			cgroupParent:      opt.CgroupParent,
-			llbCaps:           opt.LLBCaps,
-			sourceMap:         opt.SourceMap,
-			lint:              lint,
+			allDispatchStates:       allDispatchStates,
+			metaArgs:                optMetaArgs,
+			buildArgValues:          opt.BuildArgs,
+			shlex:                   shlex,
+			buildContext:            llb.NewState(buildContext),
+			proxyEnv:                proxyEnv,
+			cacheIDNamespace:        opt.CacheIDNamespace,
+			buildPlatforms:          platformOpt.buildPlatforms,
+			targetPlatform:          platformOpt.targetPlatform,
+			extraHosts:              opt.ExtraHosts,
+			shmSize:                 opt.ShmSize,
+			ulimit:                  opt.Ulimits,
+			cgroupParent:            opt.CgroupParent,
+			llbCaps:                 opt.LLBCaps,
+			sourceMap:               opt.SourceMap,
+			lint:                    lint,
+			dockerIgnoreMatcher:     dockerIgnoreMatcher,
+			copyIgnoredCheckEnabled: opt.Client.CopyIgnoredCheckEnabled,
```

Comment:
just set the matcher to `nil` if this is not enabled (avoid calling `DockerIgnorePatterns()`).

Also add a comment that this is temporary opt-in, to be removed in next release.